### ISSUE TITLE
Crash when applying battle animation flash after loading savegame. 

### ISSUE THREAD

**Issue by elsemieni** [EasyRPG/Player#2431]
**Game's name:** [The Legend of Zelda : Trident of Power ](https://pzgames.github.io/Trident-of-Power/descarga.html)
**Player's version:** 0.6.2 Android, embedded build (based on https://github.com/carlsonsantana/rpgmaker2kx2android script ), on Motorola Moto G8 Power

**Clinical case:**
Some user reported the game crashes when it loaded certain savefile. 

Sadly, there's no savefile to be shared, (in rpgmaker2kx2android, savefiles are stored in Android/data folder, unnaccesible to end-users without root privileges; also the embedded bug report feature fails gathering the files for the same reason ), so tried to get the most info possible.

However, the user was able to share a logcat about the crash:

```
01-02 21:03:08.182 31353 31385 I EasyRPG Player: Loading Save Save01.lsd
01-02 21:03:08.190 31353 31385 I EasyRPG Player: Savegame version 620 (EasyRPG Player 0.6.2)
01-02 21:03:08.207 31353 31385 I EasyRPG Player: Loaded Map Map0985.lmu
01-02 21:03:08.207 31353 31385 I EasyRPG Player: Tree: Kingdoms War IV < Kingdoms War I < TYLISTEM
01-02 21:03:09.255 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/Acción Vacía (288x256 < 72x128 < 288x256)
01-02 21:03:09.255 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/Acción Vacía (288x256 < 72x128 < 288x256)
01-02 21:03:09.262 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/$%0908 Baltho (288x256 < 432x256 < 288x256)
01-02 21:03:09.288 31353 31385 I chatty  : uid=10293(cl.pzgames.zeldatop) SDLThread identical 23 lines
01-02 21:03:09.288 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/$%0908 Baltho (288x256 < 432x256 < 288x256)
01-02 21:03:09.288 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/Acción Vacía (288x256 < 72x128 < 288x256)
01-02 21:03:09.295 31353 31385 I EasyRPG Player: Image size out of bounds: CharSet/$%0908 Baltho (288x256 < 432x256 < 288x256)
01-02 21:03:09.799   981  3023 D DlbDap2Process: process() called [5993000] times
01-02 21:03:09.901 31353 31385 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xb0 in tid 31385 (SDLThread), pid 31353 (SDLActivity)
01-02 21:03:10.085 31434 31434 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
01-02 21:03:10.091  1398  1398 I /system/bin/tombstoned: received crash request for pid 31385
01-02 21:03:10.094 31434 31434 I crash_dump64: performing dump of process 31353 (target tid = 31385)
01-02 21:03:10.109 31434 31434 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-02 21:03:10.109 31434 31434 F DEBUG   : Build fingerprint: 'motorola/sofiar_retail/sofiar:10/QPES30.79-124-7/93db12:user/release-keys'
01-02 21:03:10.109 31434 31434 F DEBUG   : Revision: 'pvt'
01-02 21:03:10.109 31434 31434 F DEBUG   : ABI: 'arm64'
01-02 21:03:10.111 31434 31434 F DEBUG   : Timestamp: 2021-01-02 21:03:10-0600
01-02 21:03:10.111 31434 31434 F DEBUG   : pid: 31353, tid: 31385, name: SDLThread  >>> cl.pzgames.zeldatop <<<
01-02 21:03:10.111 31434 31434 F DEBUG   : uid: 10293
01-02 21:03:10.111 31434 31434 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xb0
01-02 21:03:10.111 31434 31434 F DEBUG   : Cause: null pointer dereference
01-02 21:03:10.111 31434 31434 F DEBUG   :     x0  0000007d46ef5ec0  x1  0000000000000000  x2  0000000000000000  x3  0000000000000000
01-02 21:03:10.111 31434 31434 F DEBUG   :     x4  0000000000000000  x5  0000000000000000  x6  0000000000000054  x7  0000000000000054
01-02 21:03:10.111 31434 31434 F DEBUG   :     x8  0000000000000000  x9  0000000000000002  x10 0000000000000000  x11 0000000000000000
01-02 21:03:10.111 31434 31434 F DEBUG   :     x12 0000000000000054  x13 3a56ce713b44b52c  x14 269cc277e6809f1d  x15 be41013c00000000
01-02 21:03:10.111 31434 31434 F DEBUG   :     x16 0000007d505cd030  x17 0000007d4feb45f8  x18 0000000000000000  x19 0000007d46565140
01-02 21:03:10.111 31434 31434 F DEBUG   :     x20 0000007d505ed2e8  x21 0000007d505ed300  x22 0000007d470a7030  x23 0000000000000000
01-02 21:03:10.111 31434 31434 F DEBUG   :     x24 0000007da7e80268  x25 0000007d50624b60  x26 0000007d50624b50  x27 0000007da819de20
01-02 21:03:10.111 31434 31434 F DEBUG   :     x28 0000007d4eacb18c  x29 0000007d4eacad80
01-02 21:03:10.111 31434 31434 F DEBUG   :     sp  0000007d4eacad40  lr  0000007d4fe68eb4  pc  0000007d4feb4600
01-02 21:03:10.635 31434 31434 F DEBUG   : 
01-02 21:03:10.635 31434 31434 F DEBUG   : backtrace:
01-02 21:03:10.635 31434 31434 F DEBUG   :       #00 pc 00000000002a5600  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Game_Character::Flash(int, int, int, int, int)+8) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #01 pc 0000000000259eb0  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (BattleAnimation::Update()+688) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #02 pc 00000000002d48fc  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Game_Screen::Update()+48) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #03 pc 00000000002c422c  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Game_Map::Update(MapUpdateAsyncContext&, bool)+844) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #04 pc 0000000000321a78  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Scene_Map::UpdateStage1(MapUpdateAsyncContext)+52) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #05 pc 0000000000321844  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Scene_Map::Update()+60) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #06 pc 00000000002f72a4  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Player::Update(bool)+444) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #07 pc 0000000000315f5c  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Scene::MainFunction()+188) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #08 pc 00000000002f6d64  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Player::MainLoop()+208) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #09 pc 00000000002f6c10  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (Player::Run()+216) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #10 pc 00000000002e542c  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libmain.so (SDL_main+12) (BuildId: 3ad868b504408d2857b4cd9a1e6b1973727d76bf)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #11 pc 0000000000043964  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/lib/arm64/libSDL2.so (Java_org_libsdl_app_SDLActivity_nativeRunMain+696) (BuildId: 183180be94c64d8193db1d1f5717119c8d6c3d19)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #12 pc 000000000000fbd4  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/oat/arm64/base.odex (art_jni_trampoline+228)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #13 pc 00000000001375b8  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #14 pc 000000000014608c  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+276) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #15 pc 00000000002e169c  /apex/com.android.runtime/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+384) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #16 pc 00000000002dc97c  /apex/com.android.runtime/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+912) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #17 pc 000000000059d75c  /apex/com.android.runtime/lib64/libart.so (MterpInvokeStatic+368) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.635 31434 31434 F DEBUG   :       #18 pc 0000000000131994  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_static+20) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #19 pc 00000000001c5d94  /data/app/cl.pzgames.zeldatop-N7KlecIFouFQv80Kouws6Q==/oat/arm64/base.vdex (org.libsdl.app.SDLMain.run+156)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #20 pc 00000000002b22e8  /apex/com.android.runtime/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEbb.llvm.8535840774985554900+240) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #21 pc 000000000058c1b8  /apex/com.android.runtime/lib64/libart.so (artQuickToInterpreterBridge+1012) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #22 pc 0000000000140468  /apex/com.android.runtime/lib64/libart.so (art_quick_to_interpreter_bridge+88) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #23 pc 00000000001a4fd8  /system/framework/arm64/boot.oat (java.lang.Thread.run+72) (BuildId: f2156250670a6541f1418f750b583fe0c1c21eb4)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #24 pc 0000000000137334  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_stub+548) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #25 pc 000000000014606c  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+244) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #26 pc 00000000004ac4b8  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #27 pc 00000000004ad54c  /apex/com.android.runtime/lib64/libart.so (art::InvokeVirtualOrInterfaceWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue const*)+416) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #28 pc 00000000004ed2e8  /apex/com.android.runtime/lib64/libart.so (art::Thread::CreateCallback(void*)+1176) (BuildId: 0feefa19dab065a30f5d1c9825a35263)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #29 pc 00000000000e205c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+36) (BuildId: 4cd97724e0ae0bf213b33eacb7bfbbe8)
01-02 21:03:10.636 31434 31434 F DEBUG   :       #30 pc 0000000000084af0  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 4cd97724e0ae0bf213b33eacb7bfbbe8)
````

From the logcat and the stack trace I see the following:

- It happens after loading a map in a savefile.
- It tries to apply a character flash in a battle animation. 
- Apparently tries to reference null pointers, then crashes.

I don't know if there's battle animations playing or not when the savefile loads (game creator says the must be not battle animations playing at the time of saving). I will try to see if there's other non-invasive ways to get the .lsd file from app data folder (ideas?). 

I know there's a lot of changes on battle and animations, plus some regressions, so maybe this is already fixed. However, I wanted to report it anyways if this helps in something.

**Comment by Ghabry** [comment]
The save file should be either in sdcard/easyrpg/games/GAMENANE or in sdcard/easyrpg/saves/GAMENAME.

Does this Android ROM do something strange?

The path is also shown in the logcat (when the game is starting)

**Comment by Ghabry** [comment]
Likely same as the fixed #2597

Please reopen if issue happens again.
